### DISCUSSION TITLE
2024 - What is the best options today?

### DISCUSSION THREAD

**Discussion by CameronMunroe** [marcone/teslausb#833] [Q&A]
With the Pi5 now out, and other newer boards, what are the current suggested hardware builds for this? Looking for more storage, but mostly interested in "dashcam" and "sentrymode" events. I've got myself a 2023 MY. 

Also, has anyone tried this with tailscale or wireguard? It would be neat to have this thing sync using a VPN after it found a local open wifi, imo. The VPN would then secure the connection so you could connect your TeslaUSB at work and offload as well.

**Comment by stauffenberg2020** [comment]
I am pretty interested in this. Because, my M3 is not parked in the range of my home Wifi, So I have a all time running Wifi dongle in my car. So my TeslaUsb and my home pc will be in two seperate networks and an option like tailscale would be of significant help here

**Reply by stauffenberg2020** [reply]
>Also, has anyone tried this with tailscale or wireguard? It would be neat to have this thing sync using a VPN after it found a local open wifi, imo. The VPN would then secure the connection so you could connect your TeslaUSB at work and offload as well.

I got it working alongwith tailscale on my Pi Zero 2W. 
Just followed the [Manual installation](https://github.com/marcone/teslausb/wiki/Installation) procedure by @marcone. But simply installed tailscale and configured tailscale before installing teslausb

**Comment by stauffenberg2020** [comment]
> With the Pi5 now out, and other newer boards, what are the current suggested hardware builds for this? Looking for more storage, but mostly interested in "dashcam" and "sentrymode" events. I've got myself a 2023 MY.

Pi5 cannot emulate USB3.0 OTG. So I am not sure what is the extra benefit that it can bring to teslausb. 

If you are looking for more storage - Are you not able to attach an external storage in Pi4 already?

**Comment by WotAGitUR** [comment]
Thanks for mentioning that @stauffenberg2020 
I did a test setup with a spare Pi Zero 2 W on my 2024 MY and the USB wouldn't stay connected and the car kept logging errors related to slow USB. Looking at the Pi5, they mention that while the USB connector can be used, when you enable OTG it uses the USB2 controller so you get reduced performance.
Looks like the best option is still the Rock Pi 4C+ (plus) - off to order one of them now.

**Reply by stauffenberg2020** [reply]
> Looks like the best option is still the Rock Pi 4C+ (plus) - off to order one of them now.

Hi @WotAGitUR, I wanted to solve the exact problem you have been facing with your Pi Zero 2W and hence bought a Rock Pi 4C+. But the power from the Glovebox USB port of my Model 3 LR doesn't seem to be enough to run my Rock Pi 4C+ (or may be some other issue). But please drop in your comment when you get it working without any issues.

**Reply by WotAGitUR** [reply]
I went back to the Pi Zero 2W and started again. Build is much better now and no longer gives errors (I've have a '24 Model Y)
But noticed that it still can't write to the USB.
Interestingly, I also tried it plugged into my Mac. Everything shows up as expected, but if I try copying saved dash cam files back to the TeslaUSB it just disconnects.
So I'm pretty sure this isn't a car usb port issue, this is specific to the OTG usb.
But at least I have a method of testing that doesn't involve plugging it into the car every time :)

**Reply by joq3** [reply]
> Pi Zero 2W

Did you get this working?
Is Pi Zero 2W working without the Slow USB errors?

What about boot time?

**Reply by WotAGitUR** [reply]
The Pi Zero 2W is working for me, but does still have the slow USB errors.
For power, I used a PiSugar 1200mAh UPS hat which seems to work really well. That setup needed a usb splitter to power the battery and also connect the Pi Zero to the car.

**Comment by vladmihociu** [comment]
I have a few questions:
1. where are you placing this device on a MY 2023+?
2. Is the Tesla USB from glovebox connected to this PI?

**Reply by stauffenberg2020** [reply]
> 1. where are you placing this device on a MY 2023+?

You would remove the original USB supplied by Tesla in the glovebox and replace it with this SBC (Pi0 2W or any other SBC)
> 2. Is the Tesla USB from glovebox connected to this PI?

No, The original USB supplied by Tesla is not connected to this Pi naturally, But you may want to use RPi4 and connect this original USB supplied by Tesla to the RPi4 as an external device to the RPi4

**Reply by vladmihociu** [reply]
Thanks!

I was thinking that the PI would copy from the TESLA USB drive to the SD or maybe directly to the local folder.

**Comment by strickersuk** [comment]
Just looking at options myself including a Waveshare UPS. Any suggestions on a case that will take a Pi Zero 2W and a UPS HAT that sits on the pogo connectors?

**Comment by dynm** [comment]
#788 I use RockPi E for about 2 years without issues, it supports USB3 and has compact size.
